astropy/modeling/convolution.py

```diff
@@ -0,0 +1,70 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""Convolution Model"""
+# pylint: disable=line-too-long, too-many-lines, too-many-arguments, invalid-name
+import numpy as np
+
+from .core import CompoundModel, SPECIAL_OPERATORS
+
+
+class Convolution(CompoundModel):
+    """
+    Wrapper class for a Convolution model:
+
+    Parameters
+    ----------
+    TODO: fill this in
+
+
+    Notes
+    -----
+    This is wrapper is necessary to handle the limitations of the
+    pseudospectral convolution binary operator implemented in
+    astropy.convolution under `convolve_fft`. In this `convolve_fft` it
+    is assumed that the inputs `array` and `kernal` span a sufficient
+    portion of the support of the functions of the convolution.
+    Consequently, the Compound Created by the `convolve_models` function
+    makes the assumption that one should pass an input array that
+    sufficiently spans this space. This means that slightly different
+    input arrays to this model will result in different outputs, even
+    on points of intersection between these arrays.
+
+    This issue is solved by requiring a bounding_box together with a
+    resolution so that one can pre-calculate the entire domain and then
+    (by default) cache the convolution values. The function then just
+    interpolates the results from this cache.
+    """
+
+    def __init__(self, model, kernal, bounding_box, resolution, cache=True):
+        super().__init__('convolve_fft', model, kernal)
+
+        self.bounding_box = bounding_box
+        self._resolution = resolution
+        self._domain = None
+
+        self._cache_convolution = cache
+        self._cache = None
+
+    @property
+    def domain(self):
+        # TODO: fix for higher than 1D
+        if self._domain is None:
+            self._domain = np.arange(self.bounding_box[0],
+                                     self.bounding_box[1], self._resolution)
+
+        return self._domain
+
+    @property
+    def cache(self):
```

Comment:
I didn't know there is caching in `modeling` or is this a new feature? Either way, probably needs a way to invalidate it, if that doesn't exist already.

Reply:
Right now this is just a feature for the convolution, as the method employed here will compute the cached convolution exactly the same every time the model is evaluated (with the same parameters). Since this operation is relatively expensive it makes sense to store the values for reuse.

I will add some stuff to allow for the cache to be invalidated and recomputed. Same for the stashed interpolation function (currently storing that because initially computing the interpolation is expensive, but once this has been computed, it is relatively inexpensive to use).

Reply:
This has been implemented.